client/state/initial-state.js

```diff
@@ -169,20 +169,41 @@ function getInitialPersistedState( initialReducer, currentUserId ) {
 
 	if ( 'development' === process.env.NODE_ENV ) {
 		window.resetState = () => clearPersistedState().then( () => window.location.reload( true ) );
-
 		if ( shouldAddSympathy() ) {
 			// eslint-disable-next-line no-console
 			console.log(
 				'%cSkipping initial state rehydration. (This runs during random page requests in the Calypso development environment, to simulate loading the application with an empty cache.)',
 				'font-size: 14px; color: red;'
 			);
-
 			clearPersistedState();
+
+			const isStateRandomlyCleared = ! config.isEnabled( 'force-sympathy' );
```

Comment:
if I understand this properly we're _always_ setting the notification here even when persisted state _wasn't_ cleared, unless someone intentionally disables the sympathy setting.

I was originally thinking more of setting some value in `initialState` if we happened to wipe out the persisted state, and then after app boot display the notice. this would both be easier to test and also should trigger the notification to appear rather than to be evidently there from the beginning (losing the flash that draws attention).

that is, we can dispatch the notification from normal code somewhere in a normal React component added in DEV environments.

the same would work if we modify the server code to send the signal to clear persistence. if we want the ENV badge to display that then I guess we don't need to interact with Redux, or maybe we have some examples of settings already being set for the ENV badge and can follow their example.

it's okay to modify the server and have it perform the random test: "should the client wipe out persistence on this page load?" and then have the client respond to it.

Reply:
I agree with Dennis here. From a devex standpoint, it makes more sense to display the notice later, after the page has fully loaded, so it will draw attention. 

One way to achieve that could be on the dev environment to set a flag in the app context and later read that flag, and if it's set, show the notice.

Reply:
Hi @dmsnell I moved the state to a flag and dispatched a notice when the layout was mounted. Can you please have another look.